### ISSUE TITLE
are you succeed to run it?

### ISSUE THREAD

**Issue by oneOfThePeople** [cjnolet/text-renderer#1]
I mean what you do with all the missing data ? (.mat file)
Thanks

**Comment by rayush7** [comment]
@jaderberg @cjnolet @oneOfThePeople Yes even I am stuck on this because we don't know what are these .mat files. Please explain what can be done.

**Comment by oneOfThePeople** [comment]
hi,
i do it a long time ago but i try to remember 
i think this [line](https://github.com/cjnolet/text-renderer/blob/8d288a36fe48fa20a775031022dbbe521f3b0656/word_renderer.py#L466) refer to matfile with path of images that you want calculate statistic on their color
so i replace it with
`with open(matfn) as f:
            self.ims = f.read().splitlines()`
so each time there is mat file you can replace it to txt file and parse it little different.
the file contain path of images to different needs see which data set fit to what

good luck

**Comment by rayush7** [comment]
Thanks @oneOfThePeople for the help. There is also a call to loadmat function in [line.](https://github.com/cjnolet/text-renderer/blob/8d288a36fe48fa20a775031022dbbe521f3b0656/word_renderer.py#L564) I am not able to understand the structure in the SVT-train.mat file (we don't have this file with us either). What is the field gt? In this case also we need to replace the mat file with txt file (containing some image paths)?

**Comment by oneOfThePeople** [comment]
same story, you need file with all path of the images that you want (like svt) and the code use it for filling the background and border
i change the function to -
`IMLIST = []

    def __init__(self, list_fn):
        for gtmat_fn in list_fn:
            with open(gtmat_fn) as f:
                self.IMLIST += (f.read().splitlines())`

**Comment by rayush7** [comment]
After making the change in this [line](https://github.com/cjnolet/text-renderer/blob/8d288a36fe48fa20a775031022dbbe521f3b0656/word_renderer.py#L467) as you suggested, self.ims becomes a python list (containing the path of the images) so now we need to change this [line](https://github.com/cjnolet/text-renderer/blob/8d288a36fe48fa20a775031022dbbe521f3b0656/word_renderer.py#L476) too. Do you suggest anything?

**Comment by oneOfThePeople** [comment]
i really don't remember anything but i see that i change it to this
```
 # im = self.ims[..., n.random.randint(0, self.ims.shape[2])]
  imfn = random.choice(self.ims)
  im = rgb2gray(n.array(Image.open(imfn)))
```

**Comment by rayush7** [comment]
By changing the code as you have mentioned, I am able to run the code but I am getting "**can't get good contrast**" in this [line](https://github.com/cjnolet/text-renderer/blob/master/word_renderer.py#L1152) and none of the images are created. Do you think this is because I haven't chosen right images for computing correct colour statistics and filling the images or again there is a bug? Could you please suggest some standard dataset whose images I should use for filling the synthetic images and computing the statistics?

**Comment by rayush7** [comment]
Alright I solved the problem. The images that one chooses for calculating various statistics needs to be chosen carefully. I picked standard dataset like ICDAR 2013 and it worked. Thanks @oneOfThePeople

**Comment by withyou1771** [comment]
@rayush7 Can you share your code and file ?

**Comment by hungpt297** [comment]
@rayush7 Can you share your code and file?
